### PR TITLE
Adds `-Credential` parameter and Truncate Warning

### LINKED ISSUE #13: Add Truncate Warning
Emit a warning if the depth of the search exceeds the depth specified for the command. This behavior aligns with other commands, such as `ConvertTo-Json`.

### PR DESCRIPTION
**Pull request by PoshAJ**
Closes #13 and #14

- [x] Update documentation for `-Credential`.
- [x] Test a643c271d547d9dbfa5cf9260e161fd5c71bf467 in multi-domain environment.

### CODE REVIEW

**Comment by PoshAJ** [comment]
- [ ] Update documentation for `-Credential`.
- [ ] Test a643c271d547d9dbfa5cf9260e161fd5c71bf467 in multi-domain environment.

**Comment by santisq** [comment]
@PoshAJ `-Credential` can only be used with `-Server` ? Is that correct ?

File: src/PSADTree/PSADTreeCmdletBase.cs [hunk 1]
```diff
@@ -86,6 +101,17 @@ protected void Push(GroupPrincipal? groupPrincipal, TreeGroup treeGroup)
         if (Recursive.IsPresent || treeGroup.Depth <= Depth)
         {
             _stack.Push((groupPrincipal, treeGroup));
+            return;
+        }
+
+        _truncatedOutput = true;
```
Comment (santisq) [inline src/PSADTree/PSADTreeCmdletBase.cs:107]: I believe this makes more sense, can't test it though. Let me know if it isn't working as expected.

File: src/PSADTree/Commands/GetADTreePrincipalGroupMembershipCommand.cs [hunk 2]
```diff
@@ -82,22 +82,23 @@ protected override void ProcessRecord()
...
         }
         finally
         {
             principal?.Dispose();
         }
 
-        WriteObject(
-            sendToPipeline: Traverse(source),
-            enumerateCollection: true);
+        _truncatedOutput = false;
```
Comment (santisq) [inline src/PSADTree/Commands/GetADTreePrincipalGroupMembershipCommand.cs:98]: - `bool` is reset on each pipeline input to support a warning per input object.
- Showing the warning first before outputting makes more sense.

File: src/PSADTree/Commands/GetADTreeGroupMemberCommand.cs [hunk 3]
```diff
@@ -31,23 +31,25 @@ protected override void ProcessRecord()
                 return;
             }
 
-            WriteObject(
-                sendToPipeline: Traverse(
-                    groupPrincipal: group,
-                    source: group.DistinguishedName),
-                enumerateCollection: true);
+            _truncatedOutput = false;
```
Comment (santisq) [inline src/PSADTree/Commands/GetADTreeGroupMemberCommand.cs:34]: - `bool` is reset on each pipeline input to support a warning per input object.
- Showing the warning first before outputting makes more sense.

**Review by santisq — COMMENTED** [review]
I think these changes make sense given the cmdlets support pipeline input. With the previous logic it would've shown the warning message just once if I'm not mistaken.

**Comment by PoshAJ** [comment]
> @PoshAJ `-Credential` can only be used with `-Server` ? Is that correct ?

Yes, as it's currently designed. We could support it separately, but it's probably fine to require `-Server`.

**Comment by santisq** [comment]
> > @PoshAJ `-Credential` can only be used with `-Server` ? Is that correct ?
> 
> Yes, as it's currently designed. We could support it separately, but it's probably fine to require `-Server`.

Ahhh because of the overloads, didn't check them 😅 Perhaps we should add a parameter set for `-Credential` only when `-Server` is used

**Comment by PoshAJ** [comment]
> > > @PoshAJ `-Credential` can only be used with `-Server` ? Is that correct ?
> > 
> > 
> > Yes, as it's currently designed. We could support it separately, but it's probably fine to require `-Server`.
> 
> Ahhh because of the overloads, didn't check them 😅 Perhaps we should add a parameter set for `-Credential` only when `-Server` is used

Yeah, we can go a couple ways with this one?
1. Parameter Sets
2. Dynamic Parameter
3. Support `-Credential` without `-Server`

File: src/PSADTree/PSADTreeCmdletBase.cs
Comment (PoshAJ) [inline src/PSADTree/PSADTreeCmdletBase.cs:1]: There is a "bug" when using `PrincipalContext` without name and `GetGroups(PrincipalContext)`. We need to use the context for `GetGroups` to support `-Credential` and setting a default value is probably the easiest way around this. It also allows us to remove the check(s) for `Server is null` and `CredentialRequiresServer`.

**Pull request merged**